### ISSUE TITLE
focInd validity

### ISSUE THREAD

**Issue by jrpool** [cvs-health/testaro#2]
Commentators report that the `focInd` test, with `revealAll` set to `true`, wrongly reports missing focus indicators on webaim.org and aaatraq.com.

**Comment by jrpool** [comment]
Investigation showed that focus indicators on webaim.org appear only after a delay. An option to wait for focus indicators has been added to the `focInd` test. If a positive wait time is specified, the test checks whether an outline has immediately appeared. If not, it waits for 100 ms and then polls for an outline every 100 ms until an outline appears or the deadline has expired. If no outline has appeared, it tests for other focus indicators.

The definitions of a valid outline or other focus indicator were also modified. Details are shown in the`tests/focInd.js` file.

With the above changes, if a suitable wait limit is specified an element with a delayed focus indicator on webaim.org is recognized as having a focus indicator.

The revised version produces a more complex result when it tests aatraq.com. That page makes extensive use of borders as focus indicators, such as in the lists of links in the footer. The borders appear after a delay. The test recognizes those borders if the wait limit is 150 ms or less, i.e. if `focInd` checks twice. But there are some anomalies:

1. With a wait limit of 250 ms or more, i.e. if `focInd` tests three times or more, elements are reported to have no borders after being focused.
2. Whatever the wait limit is, if logging statements are inserted to show iterative results then all border data are identical in all iterations for any element. Thus, if the wait limit is 150, causing 2 iterations, both show a positive `borderTopWidth` value, and, if the wait time is 250, causing 3 iterations, then all 3 show a 0 `borderTopWidth` value.
3. If the wait limit is set to 0, the borders are recognized.

Validation with the `validation/executors/tests` module includes running `focInd` on two document files, at `validation/tests/targets/focInd`. In both cases, all validation tests are passed. The validation of the `bad.html` document, if logging statements are inserted, shows that a delayed focus indicator is not recognized until the first iteration after the delay, as expected. If the `good.html` and `bad.html` files constitute good validators, then `focInd` in working as expected. But it does not do what is expected on the aaatraq.com home page. If the reason can be discovered, then the validator files can be enriched to cover the case(s) exemplified by aaatraq.com.

**Comment by jrpool** [comment]
Although anomalies described above with the aaatraq.com home page have not been diagnosed, this issue is being closed. A related issue will be opened if it becomes known how to make `focInd` more robust with respect to that page.
